### ISSUE TITLE
[Improvement] Investigate and improve performance

### ISSUE THREAD

**Issue by MartinJohns** [MartinJohns/vscode-inline-types#6]
**Description:**
The extension should not slow down VS Code, therefor the extension should be profiled and potential performance problems should be removed.

**Comment by axetroy** [comment]
It's should be easy to improve performance.

wrap ``updateDecorations`` into an debouce function with ``lodash.debouce``.

then, It will not apply ``updateDecorations`` every time you type.

**Comment by MartinJohns** [comment]
For that I introduced the `updateDelay` (#5) with version 0.2.0, which by default is set to 0ms (no delay). I plan to actually profile the extension and see where potential bottlenecks occur and what I can do about them.

My biggest worry is the TypeScript parsing. I did not find a way to access the already used TypeScript compiler of the TypeScript extension, so I need to create an own instance of the compiler and parse the source files. That introduces an overhead, as all code is essentially parsed twice, especially with larger projects. Currently this all happens in the extension in VS Code, but it could be an option to split it up to a client/server part as demonstrated [here](https://code.visualstudio.com/docs/extensions/example-language-server). That way the TypeScript parsing would at least be in its own process and would not bother the VS Code process.
